### PR TITLE
FE: Fix topics unchecking w/ "Show Internal Topics"

### LINKED ISSUE #3255: [Topics]  All topics are unchecking with Toggling off Show Internal Topics
**Describe the bug** (Actual behavior)
All topics checking (unchecking) with toggling off(on) the Show internal Topics

**Expected behavior**
In case of toggling the Show internal Topics, the Topics should not becomes checked(unchecked) as the Internal topics are not checkable in any cases

**Set up**

https://www.kafka-ui.provectus.io/

**Steps to Reproduce**

1. Login to Kafka-ui
2. Navigate to Topics
3. Make sure Show Internal Topics is toggled on
4. Check all topics with on top checkbox
5. Toggle off the Show Internal Topics

https://user-images.githubusercontent.com/104780608/214019758-7d8d3fa9-d8c3-4df6-adf6-34bc4f5193a7.mov

### PR DESCRIPTION
**Pull request by NeiruBugz**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)

**What changes did you make?** (Give an overview)
Resolves #3255 

- fix row selection reset
- update row id getter for Table component

**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [x] Manually (please, describe, if necessary)
- [ ] Unit checks
- [ ] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by David-DB88 — APPROVED** [review]

File: kafka-ui-react-app/src/components/common/NewTable/Table.tsx [hunk 1]
```diff
@@ -160,6 +157,9 @@ const Table: React.FC<TableProps<any>> = ({
       pagination: getPaginationFromSearchParams(searchParams),
       rowSelection,
     },
+    getRowId: (originalRow, index) => {
+      return originalRow.name ? `${originalRow.name}` : `${index}`;
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/components/common/NewTable/Table.tsx:161]: Do we need the to put it in a string quotation string  literals here for the `${originalRow.name}`?

Comment (NeiruBugz) [reply kafka-ui-react-app/src/components/common/NewTable/Table.tsx:161]: I think not, it is obviously a string value)

Comment (Mgrdich) [reply kafka-ui-react-app/src/components/common/NewTable/Table.tsx:161]: Isn't it already a name?

Comment (NeiruBugz) [reply kafka-ui-react-app/src/components/common/NewTable/Table.tsx:161]: Yes, it's a string, so we do not need quotation marks - update in a moment

**Review by Mgrdich — APPROVED** [review]

**Pull request merged**